### PR TITLE
Refs #7978 fixing regex to take in account IN operator

### PR DESCRIPTION
**Pull request by pedrofs**
This is a shoot for fixing #7978 

The final regex is slightly different from the one @dereuromark has suggested.

I also tried to add some tests to catch the issue.

Let me know about any further improvement.

### CODE REVIEW

File: lib/Cake/Test/Case/Model/ModelReadTest.php [hunk 1]
```diff
@@ -87,6 +87,112 @@ public function testFetchingNonUniqueFKJoinTableRecords() {
...
+				'Product' => array(
+					'id' => 2,
+					'name' => "Silly Puddy",
+					'type' => 'Toy',
+					'price' => 3
+				)
+			),
+		);
+
+		$result = $Product->find('all', array('conditions' => array('Product.id IN' => array(1,2))));
```
Comment (dereuromark) [inline lib/Cake/Test/Case/Model/ModelReadTest.php]: space after 1, missing

**Comment by dereuromark** [comment]
Nice!

**Comment by dereuromark** [comment]
Looks like the [] array sxntax issues ( https://travis-ci.org/cakephp/cakephp/jobs/105504687 ) are not your fault, they have been there already.

**Comment by phpnut** [comment]
@dereuromark the array syntax was corrected in this PR #8124

**Comment by markstory** [comment]
Nice work @pedrofs

**Comment by pedrofs** [comment]
Thanks to @dereuromark who really found the issue. I got the easy part :)

Anyway thanks @markstory

**Comment by ionas** [comment]
:clap:

**Pull request merged**